### DISCUSSION TITLE
Deploying to Vercel - issues with connecting to nhost

### DISCUSSION THREAD

**Discussion by thomas-cruz** [nhost/nhost#2825] [Q&A] [answered]
Has anyone else faced this issue?  
I am facing some issues with deploying my project to Vercel. At first it was an issue with the pnpm-lock.yaml being out of date but when updated it would result with this error on the deplpyed Vercel site:
`Error: No Nhost client found. Use provideNhostClient() if you are outside of a component setup.`
This is a Nuxt monorepo, not sure if that matters.  This isn't a new project, it was running and connecting to nhost without any issues before updating the lockfile.
Runs fine locally, not sure what steps to take to resolve this.  

Edit: I got in touch with Vercel support. They suggested I run
```
vercel build --prod
vercel deploy --prebuilt --prod
```
Unfortunately still the same result, cannot connect to nhost.

**Comment by onehassan** [comment]
Are you using the `useNhostClient()` composable in your code ?

**Reply by thomas-cruz** [reply]
Yup. Also everything was running totally fine before updating the pnpm-lock.yaml. Used to connect to nhost without issue.

**Reply by onehassan** [reply]
Is that the only error you see on Vercel ?

**Reply by thomas-cruz** [reply]
Just checked. After updating the lockfile all the build errors went away. Currently no other errors on Vercel.

**Reply by onehassan** [reply]
How did you update the lockfile?

**Reply by thomas-cruz** [reply]
Just ran pnpm install. I also updated the packages to the latest versions with pnpm up.

**Reply by thomas-cruz** [reply]
I also ran pnpm install --fix-lockfile but somehow that resulted in a lockfile that was out of sync, running pnpm install again resolved that error.

**Reply by thomas-cruz** [reply]
I tried that and same result 😢. I also checked if Vercel would try to build a lockfile if I just remove the lockfile completely but that also didn't work.

**Reply by onehassan** [reply]
And this was working before on vercel right? - can you revert the latest commits to the lockfile? I think you need to just need to run `pnpm install` and not `pnpm install --fix-lockfile`

**Reply by thomas-cruz** [reply]
Unfortunately now it results in a build error, pnpm-lock,yaml is not up-to-date with package.json. Which is fixed by pnpm install which leads to the problem I'm having currently 😞

**Reply by onehassan** [reply]
Can you create a `.npmrc` at the root of the project with `auto-install-peers=true`, then delete the `pnpm-lock.yaml` file and then run `pnpm install`

**Reply by thomas-cruz** [reply]
Turns out I already had a .npmrc with auto-install-peers=true but I tried recreating the pnpm-lock.yaml again through pnpm install and I still have the error no nhost client found.

**Reply by onehassan** [reply]
Can you share what your root `package.json` looks like ?

**Reply by thomas-cruz** [reply]
I've made a gist [here](https://gist.github.com/thomas-cruz/2122ebdcfe62a22280b17e92c6a36d39) actually contains two package.json files, one of them for the project with all the base components and the other for the site.

**Reply by onehassan** [reply]
Just to double check, does the project build fine locally ? Also what is the version of pnpm you have on your machine ?

**Reply by thomas-cruz** [reply]
pnpm is using version 9.0.6

**Reply by thomas-cruz** [reply]
https://github.com/vercel/vercel/issues/11607
Might be a related issue? Going to try to downgrade my pnpm version

**Comment by thomas-cruz** [comment, marked as answer]
Resolved with downgrading node and pnpm versions and running pnpm install again  
added this line to the root package.json  
`  "packageManager": "pnpm@8.6.0"`  
used pnpm version 8.6.0, used node version 18.20.4
